packages/contracts/src/tx-util.ts

```diff
@@ -21,18 +23,41 @@ export class TxUtil {
     account: Account,
     option?: Tx,
   ): Promise<string> {
-    const [gas, gasPrice] = await Promise.all([
-      tx.estimateGas({
-        ...option,
+    let gasPrice: string
+    let selectedGas: number
+
+    // To Prevent using `option.gas` lower than estimated gas
+    if (!option?.gas) {
```

Comment:
No matter what the estimated as will be needed, it might be cleaner to load/store it outside of the if statements like this:

```
const estimatedGas = await tx.estimateGas({
  ...option,
  gas: undefined,
  from: account.address
})
if (!option?.gas) {
  selectedGas = estimatedGas
} else if (Fp.from(option.gas).toNumber() < estimatedGas) {
  logger.info('Warning, provided gas lower than estimated gas')
  selectedGas = estimatedGas
} else {
  selectedGas = Fp.from(option.gas).toNumber()
}
```

Reply:
I also wonder if it's worth checking to see if provided gas is lower than estimated gas? That's an extra query to the rpc node and the values we set should be sufficient for all cases. Maybe leave a comment saying `// Check gas price if some transactions are reverting` so we know what to check later if some transactions fail?

Reply:
I thought that too about  "Checking lower than estimated gas" like you. "Is it worth it?", so I just added it then waited for a response.
It would more proper way to handle this by providing comments when Tx failed.
Thank you for the nicer code!